### DISCUSSION TITLE
Why does the thumbnail method need the path?

### DISCUSSION THREAD

**Discussion by StefanOltmann** [lopcode/vips-ffm#116] [Q&A] [answered]
Hi,

thank you for making this library. I will definitely give it a try and see how it compares to skiko. It sounds all very promising.

Looking at the sample code I wonder why it takes a path parameter.
This must not be something else than the source image has, right? Why is that there?

Kind regards,

Stefan

**Comment by lopcode** [comment]
Hey Stefan 👋 Sorry, I'm not quite understanding - which path parameter specifically, is it in one of these files https://github.com/lopcode/vips-ffm/tree/main/sample/src/main/kotlin/vipsffm/sample ?

They're a mixture of showing common image handling patterns, and testing, so not all of them will make sense for your use case.

**Reply by lopcode** [reply, marked as answer]
Never mind, I see what you mean, that does look weird! The answer is you don't have to pass the path, it's an oddity caused by the automatic code generation - ideally that method would be static like `VImage.thumbnail(<path>)`, but it currently isn't. libvips offers `thumbnailImage` as an operation on an existing image, which makes more sense.

I've updated the README and sample here to use the `thumbnailImage` operation instead: https://github.com/lopcode/vips-ffm/pull/117

**Comment by StefanOltmann** [comment]
Great, thank you! 🙂
